File: 1-two-sum/two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    // for(let i=0;i<nums.length;i++){
    //     for(let j=i+1;j<nums.length;j++){
    //         if(nums[i]+nums[j]===target){
    //             return [i,j];
    //         }
    //     }
    // }

    let map=new Map();
    for(let i=0;i<nums.length;i++){
        let required=target-nums[i];
        if(required in map){
            return [i,map[required]];
        }
        map[nums[i]]=i;
    }
};

File: 1-two-sum/two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        prefix_sum={}
        for i in range(len(nums)):
            required=target-nums[i]
            if required in prefix_sum:
                return [prefix_sum[required],i]
            prefix_sum[nums[i]]=i